### DISCUSSION TITLE
Battery Low not triggered or triggering? Action does not do anything even In debug log?

### DISCUSSION THREAD

**Discussion by udochrist** [andrew-codechimp/HA-Battery-Notes#2419] [Q&A] [answered]
Hi, 
 maybe I have a misunderstanding of how this integration works when it comes to reporting:
 I have a door lock using batteries for which the device signalled a "low" state.

![Screenshot 2024-11-21 at 14 08 28](https://github.com/user-attachments/assets/a159aac4-607c-4bed-b8b3-ef866851f31e)

I'm using an automation at 9 am (from the blueprints) to trigger the "Battery notes check Battery low" check and a further "official" blueprint to generate notifications for the low batteries.

![Screenshot 2024-11-21 at 14 17 58](https://github.com/user-attachments/assets/023aa096-6931-4a50-8d60-f26d6fc3aed7)

However I did not receive any notification and cannot find any info in the log file even with the debug level set on the integration.

also calling the check action from the developer menu does not show anything in the log. 
running both automations manually is showing the traces of their run but no notification of the low battery.

Am I getting something completely wrong or might there be a bug in my setup or the integration?

regards
Udo

**Comment by andrew-codechimp** [comment]
I've just tried emulating this and not had it fail.

I think your problem may be that you haven't added the device you are expecting to check to Battery Notes, that entity icon isn't right.
Go into Integrations/Battery Notes
Add a device 
Choose Device
Select your door lock
Confirm battery types etc

Once added run the Check battery low action from developer tools

The triggers will only work on devices that have a battery notes entry added as this is what adds the extra data/monitoring.

**Comment by udochrist** [comment]
Hi Andrew,
 thanks for responding this fast!!
 I've removed the current door lock battery setup and recreated it as:

![Screenshot 2024-11-21 at 15 23 48](https://github.com/user-attachments/assets/549e42b5-61d0-419e-9809-423a99b567e3)

I've given 3 AA batteries but did not provide a template. The integration successfully found it to be eq-3.

selecting the device from the battery notes integration page gets me:
![Screenshot 2024-11-21 at 15 26 57](https://github.com/user-attachments/assets/5959ca8c-1657-44b5-9c65-3172647fcb0f)

this is not different to before. 
Running the "Battery Check" action does not do anything noticeable.

I ran it from "developer" and from the automation for good measure.
![Screenshot 2024-11-21 at 15 30 22](https://github.com/user-attachments/assets/83859520-a20d-4ce8-98b4-2472ef72885f)

No notification generated

**Comment by andrew-codechimp** [comment, marked as answer]
Ah, I see now.
That device doesn't have a battery percentage, just the battery low, so Battery Notes can't work out how to determine the battery is low automatically.  When this happens a Battery+ entity is not added and you won't get the triggers.

What you need to do is go into the Battery Notes integration, press configure on this device and within the battery low template section add something like one of these two (I think the the first as I'm guessing it's a boolean translated as Low within the UI)
Try them out in the dev tools template editor to be sure, you need to get True/False as output.
```
{{ states('sensor.mysensor_battery_low') }}
{{ states('sensor.mysensor_battery_level') == "Low" }}
```

This is detailed here
https://andrew-codechimp.github.io/HA-Battery-Notes/#battery-low-template

**Comment by udochrist** [comment]
Thanks for that. It was exacly the issue. since all of the devices already had a battery icon I thought some magic/config database would have created the needed "battery low" configs. I've now re-done about 40 of the battery notes devices with the correct templates and stuff is working! in this case its a "binary_sensor.device_name_battery" == "On"

**Reply by andrew-codechimp** [reply]
Ouch, that's a lot of devices without a proper battery percentage you have!
Unfortunately I have no magic to auto detect the low entities, they differ by manufacturer, have obscure names and different possible options.
Glad we got to the bottom of this and I've updated the documentation to make this clearer.
